// @trimble-oss/modus-web-components/dist/cjs/modus-navbar-apps-menu_4.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-c9ded160.js');

const modusNavbarAppsMenuCss = "@import url(\"https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700\");.apps-menu{background-color:#fff;border-radius:0.125rem;box-shadow:0 0 8px rgba(36, 35, 45, 0.3);display:flex;flex-direction:column;max-width:384px;padding:1.5rem;position:absolute;right:0;top:44px;z-index:100;align-items:flex-start;cursor:default;width:max-content}.apps-menu.reverse{left:0}.apps-menu .app{border-radius:0.125rem;cursor:default;display:flex;padding:0.25rem;width:100%}.apps-menu .app:hover{background-color:#e0e1e9;cursor:pointer}.apps-menu .app:active{background-color:#b7b9c3}.apps-menu .app:not(:last-child){margin-bottom:0.5rem}.apps-menu .app img{height:48px;margin-right:0.75rem;width:48px}.apps-menu .app .right{align-self:center}.apps-menu .app .right .name{font-size:0.875rem;font-weight:600}.apps-menu .app .right .description{font-size:0.75rem}";

let ModusNavbarAppsMenu = class {
  constructor(hostRef) {
    index.registerInstance(this, hostRef);
  }
  clickAppHandler(app) {
    window.open(app.url, '_blank');
  }
  render() {
    var _a;
    const direction = this.reverse ? 'reverse' : '';
    return (index.h("div", { class: `apps-menu ${direction}`, onClick: (event) => event.preventDefault() }, (_a = this.apps) === null || _a === void 0 ? void 0 : _a.map((app) => index.h("div", { class: "app", onClick: () => this.clickAppHandler(app) }, index.h("img", { src: app.logoUrl }), index.h("div", { class: "right" }, index.h("div", { class: "name" }, app.name), app.description ? index.h("div", { class: "description" }, app.description) : null)))));
  }
};
ModusNavbarAppsMenu.style = modusNavbarAppsMenuCss;

const modusNavbarMainMenuCss = "@import url(\"https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700\");.main-menu{background-color:#fff;border-radius:0.125rem;box-shadow:0 0 8px rgba(36, 35, 45, 0.3);display:flex;flex-direction:column;max-width:384px;padding:1.5rem;position:absolute;right:0;top:44px;z-index:100;border-radius:0 0 0.125rem 0.125rem;clip-path:inset(0 -10px -10px -10px);cursor:default;height:calc(100vh - 56px);left:-16px;max-width:616px;min-width:248px;padding:0;top:50px;width:max-content}";

let ModusNavbarMainMenu = class {
  constructor(hostRef) {
    index.registerInstance(this, hostRef);
  }
  render() {
    return (index.h("div", { class: 'main-menu', onClick: (event) => event.preventDefault() }, index.h("slot", null)));
  }
};
ModusNavbarMainMenu.style = modusNavbarMainMenuCss;

const modusNavbarNotificationsMenuCss = "@import url(\"https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700\");.notifications-menu{background-color:#fff;border-radius:0.125rem;box-shadow:0 0 8px rgba(36, 35, 45, 0.3);display:flex;flex-direction:column;max-width:384px;padding:1.5rem;position:absolute;right:0;top:44px;z-index:100;cursor:default;width:max-content}.notifications-menu.reverse{left:0}.notifications-menu .title{font-size:1.125rem;font-weight:600;margin-bottom:0.5rem}";

let ModusNavbarNotificationsMenu = class {
  constructor(hostRef) {
    index.registerInstance(this, hostRef);
  }
  render() {
    const direction = this.reverse ? 'reverse' : '';
    return (index.h("div", { class: `notifications-menu ${direction}`, onClick: (event) => event.preventDefault() }, index.h("div", { class: "title" }, "Notifications"), index.h("slot", null)));
  }
};
ModusNavbarNotificationsMenu.style = modusNavbarNotificationsMenuCss;

const modusNavbarProfileMenuCss = "@import url(\"https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700\");.profile-menu{background-color:#fff;border-radius:0.125rem;box-shadow:0 0 8px rgba(36, 35, 45, 0.3);display:flex;flex-direction:column;max-width:384px;padding:1.5rem;position:absolute;right:0;top:44px;z-index:100;align-items:center;cursor:default;padding-left:48px;padding-right:48px;width:fit-content}.profile-menu.reverse{left:0}.profile-menu .avatar,.profile-menu .initials{background-color:#e0e1e9;border-radius:50%;margin-bottom:1rem;height:64px;width:64px}.profile-menu .initials{align-items:center;display:flex;font-size:1.5rem;justify-content:center}.profile-menu .username{font-size:1.25rem;padding:0 0.25rem}.profile-menu .email{font-size:0.875rem;margin-bottom:1rem;padding:0 0.25rem}";

let ModusNavbarProfileMenu = class {
  constructor(hostRef) {
    index.registerInstance(this, hostRef);
    this.signOutClick = index.createEvent(this, "signOutClick", 7);
  }
  signOutClickHandler(event) {
    this.signOutClick.emit(event);
  }
  render() {
    const direction = this.reverse ? 'reverse' : '';
    return (index.h("div", { class: `profile-menu ${direction}`, onClick: (event) => event.preventDefault() }, this.avatarUrl ?
      index.h("img", { class: "avatar", height: "64", src: this.avatarUrl, alt: "Profile avatar" })
      : index.h("span", { class: "initials" }, this.initials), index.h("div", { class: "username" }, this.username), index.h("div", { class: "email" }, this.email), index.h("modus-button", { "button-style": "borderless" }, "Sign out")));
  }
};
ModusNavbarProfileMenu.style = modusNavbarProfileMenuCss;

exports.modus_navbar_apps_menu = ModusNavbarAppsMenu;
exports.modus_navbar_main_menu = ModusNavbarMainMenu;
exports.modus_navbar_notifications_menu = ModusNavbarNotificationsMenu;
exports.modus_navbar_profile_menu = ModusNavbarProfileMenu;
