// @trimble-oss/modus-web-components/dist/esm/modus-dropdown.entry.js
import { r as registerInstance, c as createEvent, h } from './index-12216e1c.js';

const modusDropdownCss = "@import url(\"https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700\");:host{font-family:\"Open Sans\", -apple-system, BlinkMacSystemFont, \"Segoe UI\", Roboto, \"Helvetica Neue\", Arial, \"Noto Sans\", sans-serif, \"Apple Color Emoji\", \"Segoe UI Emoji\", \"Segoe UI Symbol\", \"Noto Color Emoji\";position:relative}:host .dropdown:hover{cursor:pointer}:host .dropdown-list{display:none;max-height:200px;max-width:300px;overflow-x:hidden;overflow-y:auto;position:absolute;z-index:1}:host .dropdown-list.visible{display:block}:host .dropdown-list.top{bottom:0}:host .dropdown-list.right{top:0}:host .dropdown-list.left{right:0;top:0}";

let ModusDropdown = class {
  constructor(hostRef) {
    registerInstance(this, hostRef);
    this.dropdownClose = createEvent(this, "dropdownClose", 7);
    /** (optional) The placement of the dropdown in related to the toggleElement */
    this.placement = 'bottom';
    this.classByPlacement = new Map([
      ['top', 'top'],
      ['right', 'right'],
      ['bottom', 'bottom'],
      ['left', 'left'],
    ]);
  }
  documentClickHandler(event) {
    if (!event.defaultPrevented) {
      if (event.target.closest(`#${this.toggleElementId}`)) {
        this.visible = !this.visible;
      }
      else {
        this.visible = false;
      }
      if (!this.visible) {
        this.dropdownClose.emit();
      }
    }
  }
  componentWillLoad() {
    this.toggleElement = document.querySelector(`#${this.toggleElementId}`);
    if (!this.toggleElement) {
      throw Error('matching element not found for toggle-element-id');
    }
  }
  render() {
    const listContainerClass = `dropdown-list ${this.visible ? 'visible' : 'hidden'} ${this.classByPlacement.get(this.placement)}`;
    const left = this.placement === 'right' ? `${this.toggleElement.offsetWidth}px` : 'unset';
    const width = `${this.toggleElement.offsetWidth ? this.toggleElement.offsetWidth : 0}px`;
    return (h("div", { "aria-label": this.ariaLabel, class: "dropdown", role: "listbox" }, h("slot", { name: "dropdownToggle" }), h("div", { class: listContainerClass, style: { 'left': left, 'min-width': width } }, h("slot", { name: "dropdownList" }))));
  }
};
ModusDropdown.style = modusDropdownCss;

export { ModusDropdown as modus_dropdown };
