// @trimble-oss/modus-web-components/dist/components/modus-navbar.js
import { h, HTMLElement, createEvent, proxyCustomElement } from '@stencil/core/internal/client';
import { I as IconSearch } from './icon-search.js';
import { d as defineCustomElement$6 } from './modus-button2.js';
import { d as defineCustomElement$5 } from './modus-navbar-apps-menu2.js';
import { d as defineCustomElement$4 } from './modus-navbar-main-menu2.js';
import { d as defineCustomElement$3 } from './modus-navbar-notifications-menu2.js';
import { d as defineCustomElement$2 } from './modus-navbar-profile-menu2.js';

// eslint-disable-next-line
const IconMenu = (props) => {
  var _a, _b, _c;
  return (h("svg", { class: "icon-menu", height: (_a = props.size) !== null && _a !== void 0 ? _a : 16, width: (_b = props.size) !== null && _b !== void 0 ? _b : 16, onClick: props.onClick ? (event) => props.onClick(event) : null, viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
    h("path", { d: "M3 18H21V16H3V18ZM3 13H21V11H3V13ZM3 6V8H21V6H3Z", fill: (_c = props.color) !== null && _c !== void 0 ? _c : '#6A6976' })));
};

// eslint-disable-next-line
const IconNotifications = (props) => {
  var _a, _b, _c;
  return (h("svg", { class: "icon-notifications", height: (_a = props.size) !== null && _a !== void 0 ? _a : 16, width: (_b = props.size) !== null && _b !== void 0 ? _b : 16, onClick: props.onClick ? (event) => props.onClick(event) : null, viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
    h("path", { d: "M12 22C13.1 22 14 21.1 14 20H10C10 21.1 10.89 22 12 22ZM18 16V11C18 7.93 16.36 5.36 13.5 4.68V4C13.5 3.17 12.83 2.5 12 2.5C11.17 2.5 10.5 3.17 10.5 4V4.68C7.63 5.36 6 7.92 6 11V16L4 18V19H20V18L18 16Z", fill: (_c = props.color) !== null && _c !== void 0 ? _c : '#6A6976' })));
};

// eslint-disable-next-line
const IconApps = (props) => {
  var _a, _b, _c;
  return (h("svg", { class: "icon-apps", height: (_a = props.size) !== null && _a !== void 0 ? _a : 16, width: (_b = props.size) !== null && _b !== void 0 ? _b : 16, onClick: props.onClick ? (event) => props.onClick(event) : null, viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
    h("path", { d: "M4 8H8V4H4V8ZM10 20H14V16H10V20ZM4 20H8V16H4V20ZM4 14H8V10H4V14ZM10 14H14V10H10V14ZM16 4V8H20V4H16ZM10 8H14V4H10V8ZM16 14H20V10H16V14ZM16 20H20V16H16V20Z", fill: (_c = props.color) !== null && _c !== void 0 ? _c : '#6A6976' })));
};

const modusNavbarCss = "@import url(\"https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700\");nav{align-items:center;background-color:#fff;display:flex;font-family:\"Open Sans\", -apple-system, BlinkMacSystemFont, \"Segoe UI\", Roboto, \"Helvetica Neue\", Arial, \"Noto Sans\", sans-serif, \"Apple Color Emoji\", \"Segoe UI Emoji\", \"Segoe UI Symbol\", \"Noto Color Emoji\";height:56px}nav.reverse{flex-direction:row-reverse}nav .icon-menu,nav .icon-search,nav .icon-notifications,nav .icon-help,nav .icon-apps{border-radius:0.125rem;padding:0.5rem}nav .icon-menu:hover,nav .icon-search:hover,nav .icon-notifications:hover,nav .icon-help:hover,nav .icon-apps:hover{background-color:#e0e1e9}nav .icon-menu:active,nav .icon-search:active,nav .icon-notifications:active,nav .icon-help:active,nav .icon-apps:active{background-color:#b7b9c3}nav .icon-menu:active path,nav .icon-search:active path,nav .icon-notifications:active path,nav .icon-help:active path,nav .icon-apps:active path{fill:#0063a3}nav .navbar-button{position:relative}nav .navbar-button:hover{cursor:pointer}nav .left{align-items:center;display:inline-flex;padding:0 1rem}nav .left.reverse{flex-direction:row-reverse}nav .left img.product-logo{margin:0 0.5rem;max-height:32px}nav .right{align-items:center;display:inline-flex;margin-left:auto;padding:0 1rem}nav .right.reverse{flex-direction:row-reverse;margin-left:0;margin-right:auto}nav .right .search{align-items:center;display:inline-flex}nav .right .pendo-placeholder{height:40px;width:40px}nav .right .profile-menu{align-items:center;cursor:default;display:inline-flex;height:40px;justify-content:center;margin:0 0.5rem;position:relative;width:40px}nav .right .profile-menu img.avatar{background-color:#e0e1e9;border-radius:50%;height:32px;width:32px}nav .right .profile-menu .initials{align-items:center;background-color:#e0e1e9;border-radius:50%;display:flex;font-size:0.875rem;font-weight:600;height:32px;justify-content:center;width:32px}nav .right .profile-menu>:first-child:hover{border:0.25rem solid #e0e1e9}nav .right .profile-menu>:first-child:active{border:0.25rem solid #b7b9c3}nav .right .profile-menu:hover{cursor:pointer}";

let ModusNavbar$1 = class extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.__attachShadow();
    this.productLogoClick = createEvent(this, "productLogoClick", 7);
    this.profileMenuSignOutClick = createEvent(this, "profileMenuSignOutClick", 7);
  }
  documentClickHandler(event) {
    // Individual menus can prevent this listener from closing them.
    if (event.defaultPrevented) {
      return;
    }
    this.hideMenus();
  }
  signOutClickHandler(event) {
    this.profileMenuSignOutClick.emit(event);
  }
  appsMenuClickHandler(event) {
    event.preventDefault();
    this.appsMenuToggle();
  }
  appsMenuKeydownHandler(event) {
    if (event.code !== 'Enter') {
      return;
    }
    this.appsMenuToggle();
  }
  appsMenuToggle() {
    if (this.appsMenuVisible) {
      this.appsMenuVisible = false;
    }
    else {
      this.hideMenus();
      this.appsMenuVisible = true;
    }
  }
  mainMenuClickHandler(event) {
    event.preventDefault();
    this.mainMenuToggle();
  }
  mainMenuKeydownHandler(event) {
    if (event.code !== 'Enter') {
      return;
    }
    this.mainMenuToggle();
  }
  mainMenuToggle() {
    if (this.mainMenuVisible) {
      this.mainMenuVisible = false;
    }
    else {
      this.hideMenus();
      this.mainMenuVisible = true;
    }
  }
  notificationsMenuClickHandler(event) {
    event.preventDefault();
    this.notificationsMenuToggle();
  }
  notificationsMenuKeydownHandler(event) {
    if (event.code !== 'Enter') {
      return;
    }
    this.notificationsMenuToggle();
  }
  notificationsMenuToggle() {
    if (this.notificationsMenuVisible) {
      this.notificationsMenuVisible = false;
    }
    else {
      this.hideMenus();
      this.notificationsMenuVisible = true;
    }
  }
  profileMenuClickHandler(event) {
    event.preventDefault();
    this.profileMenuToggle();
  }
  profileMenuKeydownHandler(event) {
    if (event.code !== 'Enter') {
      return;
    }
    this.profileMenuToggle();
  }
  profileMenuToggle() {
    if (this.profileMenuVisible) {
      this.profileMenuVisible = false;
    }
    else {
      this.hideMenus();
      this.profileMenuVisible = true;
    }
  }
  hideMenus() {
    this.appsMenuVisible = false;
    this.mainMenuVisible = false;
    this.notificationsMenuVisible = false;
    this.profileMenuVisible = false;
  }
  render() {
    var _a, _b, _c, _d, _e, _f, _g, _h;
    const direction = this.reverse ? 'reverse' : '';
    return (h("nav", { class: direction }, h("div", { class: `left ${direction}` }, this.showMainMenu &&
      h("div", { class: "navbar-button" }, h("span", { onKeyDown: (event) => this.mainMenuKeydownHandler(event), tabIndex: 0 }, h(IconMenu, { size: "24", onClick: (event) => this.mainMenuClickHandler(event) })), this.mainMenuVisible && h("modus-navbar-main-menu", null, h("slot", { name: "main" }))), h("img", { class: "product-logo", height: "24", src: (_a = this.productLogoOptions) === null || _a === void 0 ? void 0 : _a.url, alt: "Modus navbar product logo", onClick: (event) => this.productLogoClick.emit(event) })), h("div", { class: `right ${direction}` }, this.showSearch &&
      h("div", { class: "navbar-button search" }, h(IconSearch, { size: "24" })), this.showNotifications &&
      h("div", { class: "navbar-button" }, h("span", { onKeyDown: (event) => this.notificationsMenuKeydownHandler(event), tabIndex: 0 }, h(IconNotifications, { size: "24", onClick: (event) => this.notificationsMenuClickHandler(event) })), this.notificationsMenuVisible && h("modus-navbar-notifications-menu", { reverse: this.reverse }, h("slot", { name: "notifications" }))), this.showPendoPlaceholder && h("div", { class: 'pendo-placeholder' }), this.showAppsMenu &&
      h("div", { class: "navbar-button" }, h("span", { onKeyDown: (event) => this.appsMenuKeydownHandler(event), tabIndex: 0 }, h(IconApps, { size: "24", onClick: (event) => this.appsMenuClickHandler(event) })), this.appsMenuVisible && h("modus-navbar-apps-menu", { apps: this.apps, reverse: this.reverse })), h("div", { class: "profile-menu" }, ((_b = this.profileMenuOptions) === null || _b === void 0 ? void 0 : _b.avatarUrl) ?
      h("img", { class: "avatar", height: "32", src: (_c = this.profileMenuOptions) === null || _c === void 0 ? void 0 : _c.avatarUrl, alt: "Modus navbar profile menu avatar", onClick: (event) => this.profileMenuClickHandler(event), onKeyDown: (event) => this.profileMenuKeydownHandler(event), tabIndex: 0 })
      : h("span", { class: "initials", onClick: (event) => this.profileMenuClickHandler(event), onKeyDown: (event) => this.profileMenuKeydownHandler(event), tabIndex: 0 }, (_d = this.profileMenuOptions) === null || _d === void 0 ? void 0 : _d.initials), this.profileMenuVisible &&
      h("modus-navbar-profile-menu", { "avatar-url": (_e = this.profileMenuOptions) === null || _e === void 0 ? void 0 : _e.avatarUrl, email: (_f = this.profileMenuOptions) === null || _f === void 0 ? void 0 : _f.email, initials: (_g = this.profileMenuOptions) === null || _g === void 0 ? void 0 : _g.initials, reverse: this.reverse, username: (_h = this.profileMenuOptions) === null || _h === void 0 ? void 0 : _h.username })))));
  }
  static get style() { return modusNavbarCss; }
};
ModusNavbar$1 = /*@__PURE__*/ proxyCustomElement(ModusNavbar$1, [1, "modus-navbar", {
    "apps": [16],
    "productLogoOptions": [16],
    "profileMenuOptions": [16],
    "reverse": [4],
    "showAppsMenu": [4, "show-apps-menu"],
    "showMainMenu": [4, "show-main-menu"],
    "showNotifications": [4, "show-notifications"],
    "showPendoPlaceholder": [4, "show-pendo-placeholder"],
    "showSearch": [4, "show-search"],
    "appsMenuVisible": [32],
    "mainMenuVisible": [32],
    "notificationsMenuVisible": [32],
    "profileMenuVisible": [32]
  }, [[4, "click", "documentClickHandler"], [0, "signOutClick", "signOutClickHandler"]]]);
function defineCustomElement$1() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["modus-navbar", "modus-button", "modus-navbar-apps-menu", "modus-navbar-main-menu", "modus-navbar-notifications-menu", "modus-navbar-profile-menu"];
  components.forEach(tagName => { switch (tagName) {
    case "modus-navbar":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, ModusNavbar$1);
      }
      break;
    case "modus-button":
      if (!customElements.get(tagName)) {
        defineCustomElement$6();
      }
      break;
    case "modus-navbar-apps-menu":
      if (!customElements.get(tagName)) {
        defineCustomElement$5();
      }
      break;
    case "modus-navbar-main-menu":
      if (!customElements.get(tagName)) {
        defineCustomElement$4();
      }
      break;
    case "modus-navbar-notifications-menu":
      if (!customElements.get(tagName)) {
        defineCustomElement$3();
      }
      break;
    case "modus-navbar-profile-menu":
      if (!customElements.get(tagName)) {
        defineCustomElement$2();
      }
      break;
  } });
}

const ModusNavbar = ModusNavbar$1;
const defineCustomElement = defineCustomElement$1;

export { ModusNavbar, defineCustomElement };
