// @trimble-oss/modus-web-components/dist/components/modus-progress-bar.js
import { HTMLElement, h, proxyCustomElement } from '@stencil/core/internal/client';

const modusProgressBarCss = "@import url(\"https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700\");.modus-progress-bar{display:flex;font-family:\"Open Sans\", -apple-system, BlinkMacSystemFont, \"Segoe UI\", Roboto, \"Helvetica Neue\", Arial, \"Noto Sans\", sans-serif, \"Apple Color Emoji\", \"Segoe UI Emoji\", \"Segoe UI Symbol\", \"Noto Color Emoji\";height:1rem;outline:0.0625rem solid #6a6e79;width:100%}.modus-progress-bar.default-background-color{background-color:#fff}.modus-progress-bar .progress{display:flex;font-size:0.75rem;justify-content:center;overflow:hidden;transition:width 0.6s ease}.modus-progress-bar .progress.default-color{background-color:#0063a3}.modus-progress-bar .progress.default-text-color{color:#fff}";

let ModusProgressBar$1 = class extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.__attachShadow();
    /** (optional) The progress bar's maximum value. */
    this.maxValue = 100;
    /** (optional) The progress bar's minimum value. */
    this.minValue = 0;
    /** (optional) The progress bar's value. */
    this.value = 0;
  }
  getProgressStyle(percentage) {
    const progressStyle = {
      backgroundColor: this.color,
      color: this.textColor,
      width: `${percentage}%`
    };
    return progressStyle;
  }
  getProgressBarStyle() {
    const progressBarStyle = {
      backgroundColor: this.backgroundColor
    };
    return progressBarStyle;
  }
  render() {
    const percentage = (this.value - this.minValue) / (this.maxValue - this.minValue) * 100;
    const progressBarBackgroundColorClass = this.backgroundColor ? '' : 'default-background-color';
    const progressColorClass = this.color ? '' : 'default-color';
    const progressTextColor = this.textColor ? '' : 'default-text-color';
    const progressBarClass = `modus-progress-bar ${progressBarBackgroundColorClass}`;
    const progressClass = `progress ${progressColorClass} ${progressTextColor}`;
    return (h("div", { "aria-label": this.ariaLabel, "aria-valuemax": this.maxValue, "aria-valuemin": this.minValue, "aria-valuenow": this.value, class: progressBarClass, role: "progressbar", style: this.getProgressBarStyle() }, h("div", { class: progressClass, style: this.getProgressStyle(percentage) }, this.text)));
  }
  static get style() { return modusProgressBarCss; }
};
ModusProgressBar$1 = /*@__PURE__*/ proxyCustomElement(ModusProgressBar$1, [1, "modus-progress-bar", {
    "ariaLabel": [1, "aria-label"],
    "backgroundColor": [1, "background-color"],
    "color": [1],
    "maxValue": [2, "max-value"],
    "minValue": [2, "min-value"],
    "text": [1],
    "textColor": [1, "text-color"],
    "value": [2]
  }]);
function defineCustomElement$1() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["modus-progress-bar"];
  components.forEach(tagName => { switch (tagName) {
    case "modus-progress-bar":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, ModusProgressBar$1);
      }
      break;
  } });
}

const ModusProgressBar = ModusProgressBar$1;
const defineCustomElement = defineCustomElement$1;

export { ModusProgressBar, defineCustomElement };
