// @trimble-oss/modus-web-components/dist/components/modus-modal.js
import { HTMLElement, createEvent, h, proxyCustomElement } from '@stencil/core/internal/client';
import { I as IconClose } from './icon-close.js';
import { d as defineCustomElement$2 } from './modus-button2.js';

const modusModalCss = "@import url(\"https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700\");.modus-modal{background-color:rgba(0, 0, 0, 0.4);color:#000;font-family:\"Open Sans\", -apple-system, BlinkMacSystemFont, \"Segoe UI\", Roboto, \"Helvetica Neue\", Arial, \"Noto Sans\", sans-serif, \"Apple Color Emoji\", \"Segoe UI Emoji\", \"Segoe UI Symbol\", \"Noto Color Emoji\";height:100%;left:0;overflow:auto;position:fixed;top:0;width:100%;z-index:1}.modus-modal.visible{display:block}.modus-modal.hidden{display:none}.modus-modal .content{background-color:#fff;border:0.0625rem solid #888;display:flex;flex-direction:column;justify-content:center;margin:15% auto;max-width:650px;min-width:375px;padding:0 1rem;width:max-content}.modus-modal .content .header{align-items:center;display:flex;font-size:1.125rem;font-weight:600;height:64px;justify-content:space-between}.modus-modal .content .header .icon-close{cursor:pointer}.modus-modal .content .header .icon-close:hover svg path{fill:#6a6e79}.modus-modal .content .header .icon-close:active svg path{fill:#252a2e}.modus-modal .content .body{height:auto;padding:1rem 0}.modus-modal .content .footer{align-items:center;display:flex;height:64px;justify-content:flex-end}.modus-modal .content .footer modus-button:first-of-type{margin:0 0.5rem}";

let ModusModal$1 = class extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.__attachShadow();
    this.closed = createEvent(this, "closed", 7);
    this.opened = createEvent(this, "opened", 7);
    this.primaryButtonClick = createEvent(this, "primaryButtonClick", 7);
    this.secondaryButtonClick = createEvent(this, "secondaryButtonClick", 7);
  }
  async close() {
    this.visible = false;
    this.closed.emit();
    return Promise.resolve();
  }
  async open() {
    this.visible = true;
    this.opened.emit();
    return Promise.resolve();
  }
  handleOverlayClick(event) {
    if (event.target.classList.contains('overlay')) {
      this.close();
    }
  }
  handlePrimaryKeydown(event) {
    switch (event.code) {
      case 'Enter':
        this.primaryButtonClick.emit();
        break;
    }
  }
  handleSecondaryKeydown(event) {
    switch (event.code) {
      case 'Enter':
        this.secondaryButtonClick.emit();
        break;
    }
  }
  render() {
    return (h("div", { "aria-hidden": this.closed, "aria-label": this.ariaLabel, class: `modus-modal overlay ${this.visible ? 'visible' : 'hidden'}`, onClick: (event) => this.handleOverlayClick(event), role: "dialog" }, h("div", { class: "content" }, h("div", { class: "header" }, this.headerText, h("div", { class: "icon-close", onClick: () => this.close() }, h(IconClose, { size: "20" }))), h("div", { class: "body" }, h("slot", null)), h("div", { class: "footer" }, this.secondaryButtonText &&
      h("modus-button", { "button-style": "outline", color: "secondary", onClick: () => this.secondaryButtonClick.emit(), onKeyDown: (event) => this.handlePrimaryKeydown(event) }, this.secondaryButtonText), this.primaryButtonText &&
      h("modus-button", { color: "primary", onClick: () => this.primaryButtonClick.emit(), onKeyDown: (event) => this.handleSecondaryKeydown(event) }, this.primaryButtonText)))));
  }
  static get style() { return modusModalCss; }
};
ModusModal$1 = /*@__PURE__*/ proxyCustomElement(ModusModal$1, [1, "modus-modal", {
    "ariaLabel": [1, "aria-label"],
    "headerText": [1, "header-text"],
    "primaryButtonText": [1, "primary-button-text"],
    "secondaryButtonText": [1, "secondary-button-text"],
    "visible": [32],
    "close": [64],
    "open": [64]
  }]);
function defineCustomElement$1() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["modus-modal", "modus-button"];
  components.forEach(tagName => { switch (tagName) {
    case "modus-modal":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, ModusModal$1);
      }
      break;
    case "modus-button":
      if (!customElements.get(tagName)) {
        defineCustomElement$2();
      }
      break;
  } });
}

const ModusModal = ModusModal$1;
const defineCustomElement = defineCustomElement$1;

export { ModusModal, defineCustomElement };
